llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -57,7 +57,18 @@ namespace {
 // associated with the operand.  Used for determining whether
 // s_waitcnt instruction needs to be emitted.
 
-enum InstCounterType { VM_CNT = 0, LGKM_CNT, EXP_CNT, VS_CNT, NUM_INST_CNTS };
+enum InstCounterType {
+  LOAD_CNT = 0, // VMcnt prior to gfx12.
+  DS_CNT,       // LKGMcnt prior to gfx12.
+  EXP_CNT,      //
+  STORE_CNT,    // VScnt in gfx10/gfx11.
+  NUM_NORMAL_INST_CNTS,
```

Comment:
I feel like I see this pattern more often:

```
STORE_CNT,

NORMAL_INST_CNTS_MAX = STORE_CNT,
````

Rather than having a "max" entry that's more like a sentinel value. I just prefer this style more than having a dedicated enum value for MAX

Reply:
I guess I could change this if required, but it doesn't really seem any better or worse to me.